### ISSUE TITLE
[BUG] Job is stuck in the "ENQUEUED" state.

### ISSUE THREAD

**Issue by AlexDenisov** [jobrunr/jobrunr#27]
**Describe the bug**

Job is stuck in the "ENQUEUED" state.

**Environment**
I'm using version 0.9.11.
I'm running on JRE / JDK 8

**To Reproduce**

Not sure I can provide easy steps to reproduce. Some more details are below.

**Expected behavior**

The job starts after being enqueued.

**Additional context**

I was just using JobRunr jobs for a couple of days and everything worked fine. One of the jobs has failed some days ago and I deleted it. Today, I fixed the underlying issue and re-queued the deleted job, however, it didn't start. I deleted it again and created another job - it didn't start as well. I was away for a couple of hours, when I got back it's still in the "enqueued" state.

I'm not sure where to start the debugging, so any hints are much appreciated.
Let me know if I can provide more information.

Thanks,
Alex.

**Comment by rdehuyss** [comment]
What StorageProvider are you using? Can you enable trace logging and post me the results?

**Comment by AlexDenisov** [comment]
I'm using MongoDB. Could you point me to how to enable trace logging?

I'll look into logs to see whether there are any exceptions or failures.

**Comment by AlexDenisov** [comment]
Meanwhile, I'm seeing this in the "Server info":

```
FirstHeartbeat | 18 minutes ago
LastHeartbeat | 16 minutes ago
```

I guess the server is down/not responding?

**Comment by AlexDenisov** [comment]
Ok, I got it:

```
ERROR FATAL - JobRunr encountered too many processing exceptions. Shutting down.
org.jobrunr.scheduling.exceptions.JobMethodNotFoundException:
jobs.JobService.javaJarJob(org.jobrunr.jobs.context.JobContext,java.lang.String)
```

That seems like some old job since the signature is not used for quite some time. Is it possible to include the job id into the exception message? That would simplify finding the offender.

**Comment by rdehuyss** [comment]
Correct, the background job server is down. Let me look into it - a `JobMethodNotFoundException` should not stop the complete `BackgroundJobServer`. Is there more logging?

**Comment by AlexDenisov** [comment]
That's the only relevant piece I see. All the jobs that have signature-not-found are in the succeeded state, so I'm wondering why are they being looked up for.

**Comment by rdehuyss** [comment]
Can you join me on Gitter?

**Comment by rdehuyss** [comment]
I found the reason: JobRunr is doing it's maintenance and is trying to delete these Succeeded jobs to clean up your database. This happens automatically for all jobs that are older than 36 hours. But, using Job Filters you can still hook into it and so it's trying to find the method to see whether it is annotated.

But, as the method does not exist anymore, this exception is thrown. I'm looking into a fix.

**Comment by AlexDenisov** [comment]
> Can you join me on Gitter?

Tell me the address :)

===

There seems to be another related issue: I dropped the failing jobs manually, restarted the server, and now I see the jobs are back. Is there any cache of sorts in the jobrunr that puts the jobs back?

**Comment by rdehuyss** [comment]
Mmmm, unless JobRunr has become self-aware and/or has transcended, there is no cache that I know off.

I would try to stop the server, drop the failing jobs and restart the server. Is that an option?

**Comment by AlexDenisov** [comment]
> I would try to stop the server, drop the failing jobs and restart the server. Is that an option?

Yep, this is what I did. That issue is gone now. Not sure if there is an issue on JobRunr side, or I messed something up (or both :) )

**Comment by rdehuyss** [comment]
Solved in release v0.9.16
